compiler/rustc_target/src/asm/arm64ec.rs

```diff
@@ -0,0 +1,185 @@
+use std::fmt;
+
+use rustc_data_structures::fx::FxIndexSet;
+use rustc_span::Symbol;
+
+use super::aarch64::target_reserves_x18;
+use super::{InlineAsmArch, InlineAsmType, ModifierInfo};
+use crate::spec::{RelocModel, Target};
+
+def_reg_class! {
+    Arm64EC Arm64ECInlineAsmRegClass {
+        reg,
+        vreg,
+        vreg_low16,
+    }
+}
+
+impl Arm64ECInlineAsmRegClass {
```

Comment:
Is there any way we can share some of this code with AArch64? Possibly calling into its type?

Reply:
Thanks for pointing that out. Removed Arm64ECInlineAsmRegClass and InlineAsmRegClass::Arm64EC.